game/txt/hlp/pennfunc.hlp

```diff
@@ -560,6 +560,88 @@ See also: ANSI, COLOR, @sockset, colorstyle, colors()
   Returns the absolute possessive pronoun - his/hers/its/theirs - for an object. The %a substitution returns the absolute possessive pronoun of the enactor.
 
 See also: obj(), poss(), subj()
+& CHAIN()
+  chain(<attribute list>, <base case>[, ..., <argN>])
+
+  Passes the base case through each [<object>/]<attribute> in the space delimited attribute list.
+  The result of each evaluation is passed to the next attribute, serving a similar purpose as fold().
+
+  Examples:
+    > &JSON`ADDNAME me=json_mod(%0,set,$.name,json(string,%1))
+    > &JSON`SET_CHA me=json_mod(%0,set,$.stats.charisma.val,%2)
+    > &JSON`SET_CHA_MODIFIER me=json_mod(%0,set,$.stats.charisma.mod,div(sub(%2,10),2))
+    > say chain(JSON`ADDNAME JSON`SET_CHA JSON`SET_CHA_MODIFIER,json(object),%N,13)
+    You say, "{"name":"One","stats":{"charisma":{"val":13,"mod":1}}}"
+
+    > &type me=farmer
+    > &FUN`HERO`GOLD me=json_mod(%0,set,$.gold,5)
+    > &FUN`HERO`SKILL`SWORD me=json_mod(%0,set,$.skill.sword,2)
+    > &FUN`HERO`SKILL me=%0
+    > &FUN`FARMER`GOLD me=json_mod(%0,set,$.gold,1)
+    > &FUN`FARMER`SKILL`FARM me=json_mod(%0,set,$.skill.farming,10)
+    > &FUN`FARMER`SKILL me=%0
+    > say chain(lattr(me/FUN`[v(type)]`**),json(object))
+    You say, "{"gold":1,"skill":{"farming":10}}"
+
+  See 'help chain2' for more examples.
+
+See also: fold(), map(), iter()
+& CHAIN2
+  Examples:
+    > &FUN`TESTING me=iter(lattr(%=`),\[TEST\] %i0: [chain(%i0`setup fun`wrap fun`storeresult %i0`validate fun`testoutput,u(%i0`args`0),u(%i0`args`1),u(%i0`args`2))],,%r)
+    > &FUN`STORERESULT me=setr(Result,%0)
+    > &FUN`WRAP me=u(me/chainfun,%0)
+    > &FUN`TESTOUTPUT me=if(%0,ansi(hg,Success),ansi(hr,Test failed\, output failed. Output was: '%q<Result>'))
+    
+    > &FUN`TESTING`1`ARGS`0 me=meow
+    > &FUN`TESTING`1`SETUP me=setq(1,woof)%0
+    > &FUN`TESTING`1`VALIDATE me=strmatch(%0,woof)
+    
+    > &FUN`TESTING`2`ARGS`0 me=meow
+    > &FUN`TESTING`2`SETUP me=setq(1,0)%0
+    > &FUN`TESTING`2`VALIDATE me=strmatch(%0,woof)
+    > th u(fun`testing)
+    [TEST] FUN`TESTING`1: Success
+    [TEST] FUN`TESTING`2: Test failed, output failed. Output was: 'meow'
+
+  See 'help chain3' for implementation examples.
+
+See also: fold(), map(), iter()
+& CHAIN3
```

Comment:
I added another example. But honestly, `chain()` shouldn't need these kinds of complicated examples. `iter()` doesn't. 
The question is, should these go in helpfiles, or should I just pull these kinds of examples out into this Pull Request - and make smaller examples? Because this is just getting towards 'how do you implement things using arrow' teritory.